configs/configs_test.go

```diff
@@ -16,30 +16,37 @@ package configs
 
 import (
 	"encoding/json"
+	"fmt"
 	"testing"
 	"time"
 )
 
 func TestJSONDuration(t *testing.T) {
-	var s struct {
-		Dur JSONDuration
+	testCases := []struct {
+		json         []byte
+		wantErr      string
+		wantDuration time.Duration
+	}{
+		{[]byte(`{"dur":"10s"}`), "", time.Second * 10},
+		{[]byte(`{"dur": 1}`), "json: cannot unmarshal number into Go struct field .Dur of type string", 0},
+		{[]byte(`{"dur": ""}`), "time: invalid duration ", 0},
 	}
-
-	a := []byte(`{"dur":"10s"}`)
-	if err := json.Unmarshal(a, &s); err != nil {
-		t.Errorf("Failed to unmarshal duration: %v", err)
-	}
-	if s.Dur.Duration != (time.Second * 10) {
-		t.Errorf("Unexpected duration, want: 10s, got: %s", s.Dur)
-	}
-
-	b := []byte(`{"dur": 1}`)
-	if err := json.Unmarshal(b, &s); err == nil {
-		t.Error("Didn't fail when unmarshalling a non-string type into JSONDuration")
-	}
-
-	c := []byte(`{"dur": ""}`)
-	if err := json.Unmarshal(c, &s); err == nil {
-		t.Error("Didn't fail when unmarshalling an empty string into JSONDuration")
+	for _, tc := range testCases {
+		t.Run(fmt.Sprintf("json: %s", tc.json), func(t *testing.T) {
+			var s struct {
+				Dur JSONDuration
+			}
+			err := json.Unmarshal(tc.json, &s)
+			if err != nil && tc.wantErr == "" {
+				t.Errorf("failed to unmarshal into JSONDuration: %s", err)
+			} else if err != nil && err.Error() != tc.wantErr {
+				t.Errorf("unexpected error: want: %q, got %q", tc.wantErr, err)
+			} else if err == nil && tc.wantErr != "" {
+				t.Errorf("json.Unmarshal didn't fail: want %q", tc.wantErr)
+			}
+			if tc.wantDuration != 0 && tc.wantDuration != s.Dur.Duration {
```

Comment:
I think you can drop the `tc.wantDuration != 0` condition and it'll still work, as zero should be the default on s.Dur.Duration.